src/core/mesh/qgsmeshlayer.cpp

```diff
@@ -925,38 +926,16 @@ int QgsMeshLayer::closestEdge( const QgsPointXY &point, double searchRadius, Qgs
   return selectedIndex;
 }
 
-QgsMeshDatasetIndex QgsMeshLayer::staticVectorDatasetIndex( int group ) const
```

Comment:
Why move this code? It clutters the diff here, and will make git blame a bit more frustrating...

Reply:
This MR introduces `closestFace`, `closestVertex` (which replace `snapOnFace` and `snapOnVertex` on top of an existing `closestEdge` method.  The existing `closestEdge` is in the middle of other functions. The idea if to regroup  `closestFace`, `closestVertex` and `closestEdge` together. That's why,  `closestEdge` is moved and this clutters the diff.

To solve it, I have moved the existing `closestEdge` first in a separate commit. This way, the diff of the `closestElement` commit is cleaner.